src/t8_schemes/t8_default/t8_dpyramid_bits.h

```diff
@@ -0,0 +1,352 @@
+ * \param[in] face      A face of \a p
+ * \param [out] face_child  The number of the matching child face*/
+int                 t8_dpyramid_face_child_face (const t8_dpyramid_t * p,
+                                                 const int face,
+                                                 const int face_child);
+
+/** Given a boundary element and a facenumber of this element, compute the boundary face
+ * \param[in] p          Input pyramid
+ * \param[in] face       The face number of an element
+ * \param[in, out]       The boundary face*/
+void                t8_dpyramid_boundary_face (const t8_dpyramid_t * p,
+                                               const int face,
+                                               t8_element_t * boundary);
+
+int                 t8_dpyramid_extrude_face (const t8_element_t * face,
+                                              t8_dpyramid_t * p,
+                                              const int root_face);
+
+/** Compare two elements. returns negativ if p1 < p2, zero if p1 equals p2
+ *  and positiv if p1 > p2.
+ *  If p2 is a copy of p1 then the elements are equal.
+ */
+int                 t8_dpyramid_compare (const t8_dpyramid_t * p1,
+                                         const t8_dpyramid_t * p2);
+
+/** Checks if two pyramids have the same coordinates, same type and same level.
+ * \param [in]  p   first input pyramid
+ * \param [in]  q   second input pyramid
+ * \return          0, if they are equal, 1 ow.
+ */
+/*int                 t8_dpyramid_is_equal (const t8_dpyramid_t * p,
+                                          const t8_dpyramid_t * q);*/
+
+/** Check wether a collection of 10 pyramids is a family in Morton order.
+ * \param [in]  fam A collection of pyramids
+ * \return      Nonzero if \a fam is a family of pyramids
+ */
+int                 t8_dpyramid_is_family (const t8_dpyramid_t ** fam);
+
+/** Copmute whether a given pyramid shares a given face with its root tree.
+ * \param [in]  p       The input pyramid
+ * \param [in]  face    A face of \a p
+ * \return              True, if \a is a subface of the pyramid root element.
+ */
+int                 t8_dpyramid_is_root_boundary (const t8_dpyramid_t * p,
+                                                  const int face);
+
+/** Compute the neighbor of p along a given face and the number of the dual face if
+ * the neighbor is inside the root pyramid. Return 0 if the neighbor is not inside, 1 ow.
+ * \param[in] p     Input pyramid
+ * \param[in, out] neigh    The neighbor of \a p
+ * \param[in] face          The face of \a p along which \a neigh is computed
+ * \param [in, out] neigh_face  The dual face
+ * */
+int                 t8_dpyramid_face_neighbor_inside (const t8_dpyramid_t * p,
+                                                      t8_dpyramid_t * neigh,
+                                                      const int face,
+                                                      int *neigh_face);
+
```

Comment:
Document this function.
What will happen to parent if p is level 0 (and therefore has no parent)